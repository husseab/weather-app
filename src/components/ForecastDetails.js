import React from "react";
import PropTypes from "prop-types";
import "../styles/forecast-details.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details_date">{formattedDate}</div>
      <div className="forecast-details__temperature_max">
        Max temperature: {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__temperature_min">
        Min temperature: {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details_humidity">Humidity: {humidity}</div>
      <div className="forecast-details__wind_speed">
        Wind Speed: {wind.speed}
      </div>
      <div className="forecast-details__wind_direction">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
