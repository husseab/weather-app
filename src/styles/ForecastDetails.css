.forecast-details {
  display: grid;
  justify-content: center;
  flex-wrap: wrap;
  padding-top: 20px;
}
