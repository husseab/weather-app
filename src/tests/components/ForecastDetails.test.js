import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      max: 22,
      min: 12,
    },
    humidity: 11,
    wind: {
      speed: 50,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details_date");
    expect(getByText("Max temperature: 22°C")).toHaveAttribute(
      "class",
      "forecast-details__temperature_max"
    );
    expect(getByText("Min temperature: 12°C")).toHaveAttribute(
      "class",
      "forecast-details__temperature_min"
    );
    expect(getByText("Humidity: 11")).toHaveAttribute(
      "class",
      "forecast-details_humidity"
    );
    expect(getByText("Wind Speed: 50")).toHaveAttribute(
      "class",
      "forecast-details__wind_speed"
    );
    expect(getByText("Wind Direction: s")).toHaveAttribute(
      "class",
      "forecast-details__wind_direction"
    );
  });
});
