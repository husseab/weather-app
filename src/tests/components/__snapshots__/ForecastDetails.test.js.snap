// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature_max"
    >
      Max temperature: 22°C
    </div>
    <div
      class="forecast-details__temperature_min"
    >
      Min temperature: 12°C
    </div>
    <div
      class="forecast-details_humidity"
    >
      Humidity: 11
    </div>
    <div
      class="forecast-details__wind_speed"
    >
      Wind Speed: 50
    </div>
    <div
      class="forecast-details__wind_direction"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
